cpp/include/raft/distance/detail/distance_ops/template.cuh

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/util/cuda_utils.cuh>
+
+namespace raft::distance::detail::ops {
+
+// Describes the computation the template distance
+//
+// Fill in the TODO items.
+
+template <typename DataT, typename AccT, typename IdxT>
+struct template_distance_op {
+  TODO member;
```

Comment:
Oh I see- so this file is just meant to be a guide for a future developer and never meant to be included or used directly. I wonder if we should should include this in the docs instead of putting it in the code. Maybe we could start creating links off of the developer guide for guides related to individual namespaces?

Reply:
Yes indeed. I agree that adding more documentation would be really great. I would suggest holding off a bit till the design solidifies following PRs #1295 and #1307.

Reply:
Can you open an issue for this and reference the issue in the code please? That way we don't lose sight of it with everything else going on.